### DISCUSSION TITLE
Releasing a new version of aiosmtpd?

### DISCUSSION THREAD

**Discussion by P-EB** [aio-libs/aiosmtpd#319] [General]
Hi,

Some fixes that would be nice to include in the next Debian release (bookworm, freeze is due to january 2023), in particular regarding Python 3.10 and 3.11.

Would you accept to consider the opportunity to make a release of aiosmtpd?

Cheers!

-- 
PEB

**Comment by waynew** [comment]
I'd certainly be happy to cut a new release - we do have several PRs open here that I think I'd like to get merged first. I'm off work this week, though today I'm driving so  probably can't do much.

But I'll try to make some time this evening to put together a plan/milestone 👍

**Comment by P-EB** [comment]
Thanks <3

**Comment by waynew** [comment]
So, my *plan* is that we should target late December. Specifically because I'm taking some PTO then. We had already scheduled some stuff for 1.5, so I'm updating the milestone and going through PRs/issues to see what we need to add. If anyone wants to make some contributions/review code/update docs, that would be :+1: :tada: :partying_face:

**Comment by pepoluan** [comment]
Because of the very close deadline I'm going to release an interim release 1.4.3 before we get to 1.5.0.

Changes should be minimal compared to 1.4.2.

This should allow time for projects depending on us to do their own integration testing, in time for the Debian freeze.

Major change might be the dropping of Python 3.6 and adding of up to Python 3.11

That said, the aiosmptd page on PyPI seems to gain "critical project" label, and I no longer can manage it. Can you update the page after 1.4.3 is packaged and released, @waynew ?

**Reply by waynew** [reply]
@pepoluan it should be updated via whatever wheel we upload, or was there something else you wanted to change? 

I also need to get you my signing key fingerprint so we can add that to the docs as well.

**Reply by pepoluan** [reply]
Just the classifiers since we're dropping 3.6.

Will that happen automatically as I up the wheel?

Also I'll sign your key later today.

**Reply by waynew** [reply]
Oh yeah - the classifiers will come from the wheel or sdist metadata 👍

**Comment by pepoluan** [comment]
If all goes well and nothing blows up today, I'll release 1.4.3 final tonight (my time = UTC+7)

**Reply by pepoluan** [reply]
## And it's released!

Yay!

**Reply by pepoluan** [reply]
Can someone inform the Debian Python team that their [Bug 1025019](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1025019) is likely fixed by `aiosmtpd-1.4.3` ?

cc @waynew @P-EB @warsaw @CharlemagneLasse

**Reply by pepoluan** [reply]
Yay it's picked up by Debian, and now 1.4.3 is on-track to be included.

All packages that depends on aiosmtpd should be able to go forward now :-)

**Reply by warsaw** [reply]
Awesome, thanks again for continuing to maintain this library!

**Comment by pepoluan** [comment]
Version 1.4.4 -- which works around the deprecation of `asyncio.get_event_loop()` -- is waiting on the wings, ready to be released.

I will let it simmer for awhile before doing a release, just in case someone try out the latest `master` and discovers some breaking issues.

If no breaking changes found, I plan on releasing 1.4.4 on Monday (2023-01-16).

But aside from that annoying DeprecationWarning, 1.4.3 should be identical to 1.4.4.

More significant changes are pushed back to 1.4.5 and 1.5.0.

**Comment by warsaw** [comment]
It's been way too long since I dug into any of this stuff, but is it possible to use [asyncio.run()](https://docs.python.org/3/library/asyncio-runner.html#asyncio.run)?

**Reply by pepoluan** [reply]
I think so 😄

I'm delaying 3.7 improvements to 1.5.0 though.

But I already have my own branch in which I'm gradually 'modernizing' the asyncio plumbing.
